### PR TITLE
Add onClear functionality

### PR DESCRIPTION
**Pull request by EWhite613**
**This project uses [semver](http://semver.org), please check the scope of this pr:**

- [ ] #none# - documentation fixes and/or test additions
- [ ] #patch# - backwards-compatible bug fix
- [x] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG
* Add `onClear` functionality to both frost-text, and frost-textarea

### CODE REVIEW

**Comment by EWhite613** [comment]
clear

**Comment by notmessenger** [comment]
I'm guessing maybe it's only specific to the dummy app but why are there two notifications displayed for the textarea demo?

File: docs/frost-text.md [hunk 1]
```diff
@@ -23,6 +23,7 @@
 | `form` | `string` | `<form-owner>` | form element that this is associated with (its form owner) |
 | `hook` | `string` | `<unique-name>` | name used for testing with ember-hook |
 | `maxlength` | `number` | `<maxlength-value>` | maximum number of characters a user can enter |
+| `onClear`| `function`/`action`| empty function | triggers associated action when `X` is clicked and text is cleared |
```
Comment (notmessenger) [inline docs/frost-text.md]: Similar documentation for `frost-button` is

onClick | string | <action-name> | triggers associated action when the button is clicked

**Comment by EWhite613** [comment]
@notmessenger looks like there was an extra notification container (Removed now)

File: docs/frost-textarea.md [hunk 2]
```diff
@@ -15,6 +15,7 @@ a text area component
 | | | `true` | disabled text area |
 | `form` | `string` | `<form-owner>` | form element that this is associated with (its form owner) |
 | `hook` | `string` | `<unique-name>` | name used for testing with ember-hook |
+| `onClear`| `function`/`action`| empty function | triggers associated action when `X` is clicked and text is cleared |
```
Comment (notmessenger) [inline docs/frost-textarea.md]: Similar documentation for `frost-button` is

onClick | string | <action-name> | triggers associated action when the button is clicked

File: addon/components/frost-text.js [hunk 3]
```diff
@@ -60,6 +61,7 @@ export default Component.extend(FrostEventsProxyMixin, {
       isClearEnabled: false,
       isClearVisible: false,
       isHookEmbedded: false,
+      onClear: () => {},
```
Comment (notmessenger) [inline addon/components/frost-text.js]: This does not match the pattern used elsewhere in the repo, such as at https://github.com/ciena-frost/ember-frost-core/blob/master/addon/components/frost-expand.js#L94

Was this deviation intentional?

File: addon/components/frost-textarea.js [hunk 4]
```diff
@@ -48,6 +49,7 @@ export default Component.extend(FrostEventsProxyMixin, {
       isClearEnabled: false,
       isClearVisible: false,
       disabled: false,
+      onClear: () => {},
```
Comment (notmessenger) [inline addon/components/frost-textarea.js]: This does not match the pattern used elsewhere in the repo, such as at https://github.com/ciena-frost/ember-frost-core/blob/master/addon/components/frost-expand.js#L94

Was this deviation intentional?

File: docs/frost-text.md (hunk 1 above)
Comment (EWhite613) [reply docs/frost-text.md]: Ya I noticed that, but `string` doesn't make sense (in both cases). Button and this both take a `function` (action helper just returns a `function`) for `onClick` and `onClear`

Comment (notmessenger) [reply docs/frost-text.md]: You cannot pass in "just a" `function` though so `closure action` is probably the better descriptor.  Once you settle on what to use please either a) update the rest of the components descriptions to match or b) open an issue identifying that this work needs to be done.

File: addon/components/frost-text.js (hunk 3 above)
Comment (EWhite613) [reply addon/components/frost-text.js]: Nope, didn't see the other pattern. I'll change it to match those

**Comment by notmessenger** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-core/pull-request/534/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-core)

File: docs/frost-button.md [hunk 5]
```diff
@@ -14,8 +14,8 @@
 |  |  | `true` | disabled button |
 | `hook` | `string` | `<unique-name>` | name used for testing with ember-hook |
 | `icon` | `string` | `<icon-name>` | name of a frost-icon |
-| `onClick` | `string` | `<action-name>` | triggers associated action when the button is clicked |
-| `onFocus` | `string` | `<action-name>` | triggers associated action when the button received focusin event |
+| `onClick` | `closure-action` | `<closure-action>` | triggers associated action when the button is clicked |
+| `onFocus` | `closure-action` | `<closure-action>` | triggers associated action when the button received focusin event |
```
Comment (ghost) [inline docs/frost-button.md]: `triggers associated action when the button receives focusin event`

**Comment by notmessenger** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-core/pull-request/534/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-core)

**Pull request merged**